src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,16 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Returns a concatenated version of the printable strings of each object
+     */
+    default String getPrintableString(Printable... printables) {
+        String result = "";
+        for (Printable attribute: printables) {
+            result = result + " " + attribute.getPrintableString();
```

Comment:
:bulb: *Java tip*: Java 8 has a `StringJoiner` class that can join multiple strings using a delimiter e.g. `aaa` `bbb` `ccc` --> `aaa, bbb, ccc` https://www.mkyong.com/java8/java-8-stringjoiner-example/